### ISSUE TITLE
Allow limiting the number of parallel instances of an @.service

### ISSUE THREAD

**Issue by georgmu** [systemd/systemd#17097]
**Is your feature request related to a problem? Please describe.**
I want to write a systemd service to update a program. Since the program itself might be updated too, this is solved using a systemd service.
The argument of the @.service is a path to the update file. To avoid that there is more than one update in parallel, I want to limit the number of instances to one. But since it is paramerized (the filename), I want to use an @.service. Otherwise, I would have to copy the file to a specific place or write an environment file, which is also prone to race conditions.

**Describe the solution you'd like**
I propose something like a MaxInstanceCount= service variable to limit the number of instances which may be active in parallel.
If more services are started, an error would be returned.

**Describe alternatives you've considered**
Currently, I solve the problem by using flock to a lock file in the shell script which is called by the service. An option in the service file would be more elegant.

**Comment by poettering** [comment]
We have this now, indirectly, via 9ca16f6f18543754e6dc1eeb54af474468e228ff.

Since we synthesize an implicit slice for template units, one can simply set the limit on the slice, and gets the desired effects.

**Comment by astro** [comment]
In my use-case I don't want to limit the total number of services but the number of services that are starting up, using sd_notify. Should I open a new issue?
